core/primitives-core/src/types.rs

```diff
@@ -23,7 +29,77 @@ pub type ShardId = u64;
 /// Balance is type for storing amounts of tokens.
 pub type Balance = u128;
 /// Gas is a type for storing amount of gas.
-pub type Gas = u64;
+#[derive(
+    Debug,
+    Default,
+    BorshSerialize,
+    BorshDeserialize,
+    Serialize,
+    Deserialize,
```

Comment:
These may need additional attributes to make sure that the serialized structure shape doesn't change.

Also 
```suggestion
    BorshSerialize,
    BorshDeserialize,
    serde::Serialize,
    serde::Deserialize,
```

as per the [style guide](https://github.com/near/nearcore/blob/master/docs/practices/style.md#derives).